scripts/binary_size_check.sh

```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-if [ "${DEBUG}" = 1 ]; then
+if [ "${DEBUG}" -eq 1 ]; then
```

Comment:
What benefit are we gaining from this change?

Reply:
We compare numbers according to the rules in bash.

Reply:
@rninov for future reference, our goal is to have the K3s scripts POSIX compliant, so we do not allow bashisms/bash-specific implementations of functionality.

For this PR and this specific change, I think this is sane as it is implementing a change from string comparison to algebraic integer comparison.